client/js/table-view.js

```diff
@@ -62,13 +63,52 @@ class TableView {
       }
       docFragment.appendChild(tr);
     }
-    removeChildren(this.bodyEle);
-    this.bodyEle.appendChild(docFragment);
+    removeChildren(this.bodyEl);
+    this.bodyEl.appendChild(docFragment);
+  }
+
+  renderTableFoot() {
+    removeChildren(this.footEl);
+    const docFragment = document.createDocumentFragment();
+    const tr = createTR();
+    for(let col = 0; col < this.model.numOfCols; col++) {
+      const colVals = this.fetchColumnVals.bind(this, col);
+      let sum = '';
+      if (colVals().length > 0) {
+        sum = this.computeColumnSum(colVals()); 
+      }
+      const td = createTD(sum.toString());
+      tr.appendChild(td);
+    }
+    docFragment.appendChild(tr);
+    this.footEl.appendChild(docFragment);
+    this.setFootColor();
+  }
+
+  fetchColumnVals(colIndex) {
+    return getTableRows()
+      .map((tr) => {
+        const cellVal = parseInt(tr.cells[colIndex].textContent, 10); 
+        return this.validateColumnVals(cellVal);
+      })
+      .filter((cellVal) => cellVal !== '');
+  }
+
+  validateColumnVals(cellVal) {
+    return (!Number.isNaN(cellVal)) ? cellVal : '';
```

Comment:
You don't need the surrounding parens, it's no big deal but less visual cruft is good.